dbus_dump_watcher.cpp

```diff
@@ -0,0 +1,141 @@
+                                 DumpType dumpType) :
+    _bus(bus),
+    _entryIntf(entryIntf), _dumpType(dumpType),
+    _intfAddWatch(std::make_unique<sdbusplus::bus::match_t>(
+        bus,
+        sdbusplus::bus::match::rules::interfacesAdded() + sender(dumpService),
+        [this](auto& msg) { this->interfaceAdded(msg); })),
+    _intfRemWatch(std::make_unique<sdbusplus::bus::match_t>(
+        bus,
+        sdbusplus::bus::match::rules::interfacesRemoved() + sender(dumpService),
+        [this](auto& msg) { this->interfaceRemoved(msg); }))
+{
+}
+
+void DBusDumpWatcher::interfaceAdded(sdbusplus::message::message& msg)
+{
+    sdbusplus::message::object_path objPath;
+    DBusInteracesMap interfaces;
+    msg.read(objPath, interfaces);
+    std::string strObjPath = objPath;
+    log<level::ERR>(
+        fmt::format("DBusDumpWatcher::interfaceAdded path ({})", strObjPath)
+            .c_str());
+    auto iter = interfaces.find(_entryIntf);
+    if (iter == interfaces.end())
+    {
+        // ignore not specific to the dump type being watched
+        return;
+    }
+    uint32_t id = std::stoul(objPath.filename());
+    _entryPropWatchList.emplace(
+        strObjPath, std::make_unique<sdbusplus::bus::match_t>(
+                        _bus,
+                        sdbusplus::bus::match::rules::propertiesChanged(
+                            strObjPath, progressIntf),
+                        [this, strObjPath, id](auto& msg) {
+                            this->propertiesChanged(strObjPath, id, msg);
+                        }));
+}
+
+void DBusDumpWatcher::interfaceRemoved(sdbusplus::message::message& msg)
+{
+    sdbusplus::message::object_path objPath;
+    DBusInteracesMap interfaces;
+    msg.read(objPath, interfaces);
+    std::string strObjPath = objPath;
+    log<level::ERR>(
+        fmt::format("DBusDumpWatcher::interfaceRemoved path ({})", strObjPath)
+            .c_str());
+    auto iter = interfaces.find(_entryIntf);
+    if (iter == interfaces.end())
+    {
+        // ignore not specific to the dump type being watched
+        return;
+    }
+    _entryPropWatchList.erase(strObjPath);
+}
+
+void DBusDumpWatcher::propertiesChanged(const std::string& objPath, uint32_t id,
+                                        sdbusplus::message::message& msg)
+{
+    std::string interface;
+    DBusPropertiesMap propMap;
+    msg.read(interface, propMap);
+    log<level::INFO>(
+        fmt::format("propertiesChanged objectpath ({}) id ({})", objPath, id)
+            .c_str());
+
+    bool fcomplete = isDumpProgressCompleted(propMap);
+    if (!fcomplete)
+    {
+        log<level::INFO>(
+            fmt::format("propertiesChanged objectpath ({}) status is not "
+                        "complete",
+                        objPath)
+                .c_str());
+        return;
+    }
+
+    uint64_t size = getDumpSize(_bus, objPath);
+    openpower::dump::pldm::sendNewDumpCmd(id, _dumpType, size);
```

Comment:
yes we do not wait, if command is failed or host is down this application also need to go for reset.

Reply:
How you will know that the PLDM command is failed and How does this app get reset? since we did not look at the response and did not throw an exception, right? 

I think you did not implement the PLDM send command, I could wait until you implemented it to think about my comments on this.

Reply:
basically the PLDM changes are in commitr 3 , if there is a failure it is raising elog exception which will be propogated to the exceotion handling code in mainapp.